packages/oauth/oauth-provider/src/dpop/dpop-manager.ts

```diff
@@ -47,110 +48,153 @@ export class DpopManager {
    * @see {@link https://datatracker.ietf.org/doc/html/rfc9449#section-4.3}
    */
   async checkProof(
-    proof: unknown,
-    htm: string, // HTTP Method
-    htu: string | URL, // HTTP URL
-    accessToken?: string, // Access Token
-  ) {
-    if (Array.isArray(proof) && proof.length === 1) {
-      proof = proof[0]
+    httpMethod: string,
+    httpUrl: Readonly<URL>,
+    httpHeaders: Record<string, undefined | string | string[]>,
+    accessToken?: string,
+  ): Promise<null | DpopResult> {
+    // Fool proofing against use of empty string
+    if (!httpMethod) {
+      throw new TypeError('HTTP method is required')
+    }
+
+    const proof =
+      Array.isArray(httpHeaders['dpop']) && httpHeaders['dpop'].length === 1
+        ? httpHeaders['dpop'][0]!
+        : httpHeaders['dpop']
+
+    if (proof === undefined) {
+      return null
     }
 
     if (!proof || typeof proof !== 'string') {
       throw new InvalidDpopProofError('DPoP proof required')
     }
```

Comment:
Should we offer a different error message if `proof` is an array?  That would indicate multiple proofs were received.